src/client/datascience/notebook/kernelWithMetadata.ts

```diff
@@ -98,17 +101,4 @@ export class VSCodeNotebookKernelMetadata implements VSCNotebookKernel {
         handlerDisposables.push({ dispose: () => statusChangeDisposable.dispose() });
         handlerDisposables.push({ dispose: () => kernelDisposedDisposable?.dispose() });
     }
-
-    private async chainExecution(next: () => Promise<void>): Promise<void> {
-        if (this.pendingExecution) {
-            try {
-                await this.pendingExecution;
-            } catch (e) {
-                // Errors are handled elsewhere
-                traceInfo(`Kernel execution previous failure: ${e}`);
-            }
-        }
-        this.pendingExecution = next();
```

Comment:
This wasn't chaining promises correctly, possible that this method got called twice.
Both would wait for previous pending execution to complete.
Then both would end up with the `next` getting executed & only one of them will have `pendingExecution` updated with the promise of `next`. Meaning both get executed in parallel.